web/packages/core/src/ruffle-player.ts

```diff
@@ -1663,7 +1668,23 @@ export class RufflePlayer extends HTMLElement {
                 if (enabled !== false) {
                     menuItem.addEventListener(
                         this.contextMenuSupported ? "click" : "pointerup",
-                        onClick,
+                        (event: MouseEvent) => {
+                            // Prevent the menu from being destroyed.
+                            // It's required when we're dealing with async callbacks,
+                            // as the async callback may still use the menu in the future.
+                            event.stopPropagation();
+
+                            const result = onClick(event);
```

Comment:
I would `await onClick` here and make the function async of course.

Reply:
Fixed. I tried to avoid changing every `onClick` callback to be asynchronous :sweat_smile: